src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -82,29 +81,93 @@ def __init__(
         >>> pd.Index(list('abc'))
         Index(['a', 'b', 'c'], dtype='object')
 
+        # Snowpark pandas casts all int types to int64 by default
         >>> pd.Index([1, 2, 3], dtype="uint8")
-        Index([1, 2, 3], dtype='uint8')
+        Index([1, 2, 3], dtype='int64')
+        """
+        self.is_lazy = convert_to_lazy
+        if self.is_lazy:
+            self.set_query_compiler(
+                data=data,
+                dtype=dtype,
+                copy=copy,
+                name=name,
+                tupleize_cols=tupleize_cols,
+            )
+        else:
+            self.set_local_index(
+                data=data,
+                dtype=dtype,
+                copy=copy,
+                name=name,
+                tupleize_cols=tupleize_cols,
+            )
+
+    def set_query_compiler(
+        self,
+        # Any should be replaced with SnowflakeQueryCompiler when possible (linter won't allow it now)
```

Comment:
turn this into a TODO and a JIRA ticket as well, to prevent us from forgetting to do this. Add ticket to TODO too, thanks!

Reply:
+1 to this and the other comment